include/openPMD/Datatype.hpp

```diff
@@ -71,8 +71,6 @@ enum class Datatype : int
 
     BOOL,
 
-    DATATYPE = HIGHEST_DATATYPE,
-
     UNDEFINED
```

Comment:
Potentially for iterating: last data type as `NUM_DATATYPES` and then it will automatically be the number of elements in the enum (because we start with `= 0`)

Reply:
But that would only really work if we did *not* assign it to `HIGHEST_DATATYPE`? Because that is defined as 1000, so you would enumerate from 0 to 1000, but we only have 30 datatypes or so.
See [here](https://stackoverflow.com/questions/14989274/is-it-possible-to-determine-the-number-of-elements-of-a-c-enum-class):
```cpp
enum class Example { A, B, C, D, E, Count };
```
I'd suggest, if we want to do such a thing, we just keep `Datatype::UNDEFINED` at the end of the enum and use that for enumerating.

Latest commit removes HIGHEST_DATATYPE and LOWEST_DATATYPE, they were only really used internally in `switchType` where it didn't make sense.